[HttpFoundation] Add a dependency on the mbstring polyfill

jakzal opened the pull request:
| Q | A |
| --- | --- |
| Branch | 2.3 |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Alternative for #18029.

jakzal committed:
[HttpFoundation] Add a dependency on the mbstring polyfill

fabpot commented on src/Symfony/Component/HttpFoundation/Tests/BinaryFileResponseTest.php:
```diff
@@ -36,6 +36,7 @@ public function testConstruction()
 
     public function testConstructWithNonAsciiFilename()
     {
+        var_dump(scandir(__DIR__.'/Fixtures/'));
```
should be removed :)

jakzal replied:
Just debugging. I'm not exactly sure why this fails on windows :)

jakzal committed:
[HttpFoundation] Fix the test on windows

A file with non-ascii characters in the name needs to be created from PHP
in order for it to be read properly on all platforms.

jakzal commented:
Tests pass everywhere now. In order for file names with non-ascii characters work on all platforms they need to be created from PHP (just like we do it in FileTest).

Status: needs review

xabbuh commented:
We can now also remove the fixtures file that is not used anymore, can't we?

jakzal commented:
It's used in the other test.

xabbuh commented on src/Symfony/Component/HttpFoundation/Tests/BinaryFileResponseTest.php:
```diff
@@ -36,7 +36,13 @@ public function testConstruction()
+
+        $this->assertSame('fööö.html', $response->getFile()->getFilename());
+
+        @unlink(sys_get_temp_dir().'/fööö.html');
```
We should remove the file before the assertion. Otherwise it wouldn't be removed when the assertion failed.

jakzal replied:
Good point.

xabbuh commented:
Hm indeed, can't we reuse the same approach in that test?

jakzal commented:
@xabbuh actually, the other test didn't need the file to exist, as only the file name is used there. I removed the fixture.

jakzal commented:
@symfony/deciders this PR, when merged, will fix the build

xabbuh commented:
:+1: 

Status: Reviewed

nicolas-grekas commented:
Thank you @jakzal.
